packages/admin/cms-admin/src/pages/pageTree/useExtractPages.tsx

```diff
@@ -0,0 +1,270 @@
+
+                    if (!documentType) {
+                        throw new Error(`Unknown document type "${documentType}"`);
+                    }
+
+                    try {
+                        const query = documentType.getQuery;
+
+                        if (query) {
+                            const { data } = await client.query<GQLPageQuery, GQLPageQueryVariables>({
+                                query,
+                                variables: {
+                                    id: page.id,
+                                },
+                                context: LocalErrorScopeApolloContext,
+                            });
+
+                            if (data?.page?.document) {
+                                const documentWithUpdateContents = documentType.replaceTextContents?.(data.page.document, content.contents);
+
+                                if (documentType.updateMutation) {
+                                    await client.mutate<GQLUpdatePageMutation, GQLUpdatePageMutationVariables>({
+                                        mutation: documentType.updateMutation,
+                                        variables: {
+                                            pageId: data.page.document.id,
+                                            input: { content: documentWithUpdateContents?.content, seo: documentWithUpdateContents?.seo },
+                                            attachedPageTreeNodeId: page.id,
+                                        },
+                                        context: LocalErrorScopeApolloContext,
+                                    });
+                                }
+
+                                const importedName = content.contents.find((item) => item.original === page.name)?.replaceWith;
+                                const importedSlug = content.contents.find((item) => item.original === page.slug)?.replaceWith.toLowerCase();
+
+                                if (importedName !== page.name || importedSlug !== page.slug) {
+                                    await client.mutate<GQLUpdatePageNodeMutation, GQLUpdatePageNodeMutationVariables>({
+                                        mutation: updatePageNodeMutation,
+                                        variables: {
+                                            nodeId: page.id,
+                                            input: {
+                                                name: importedName && importedName !== "" ? importedName : page.name,
+                                                slug:
+                                                    importedSlug && importedSlug !== ""
+                                                        ? transformToSlug(importedSlug ?? page.slug, locale)
+                                                        : page.slug,
+                                                attachedDocument: { id: data.page.document.id, type: data.page.document.__typename },
```

Comment:
There's no need to update the attached document, name and slug should be enough

Reply:
the attachedDocument is only added as it is a required parameter

Reply:
Okay, this shouldn't be required in the updatePageTreeNodeMutation (out of scope of this PR)